### DISCUSSION TITLE
Can we retry on return result?

### DISCUSSION THREAD

**Discussion by SysuJayce** [hynek/stamina#46] [Ideas]
Hi there, I'd like to retry my function on some specific return result.
For example, if func() return 0, I want to retry it, but currently stamina only allows retry on some exceptions

**Comment by hynek** [comment]
I'm not sure I want that in stamina (the goal isn't a reimplementation of Tenacity after all :)) but you simulate that quite easily by writing your own decorator that raises a well-known exception which you let stamina retry on.

Something like:

```python
class PleaseRetryError(Exception):
    pass

@stamina.retry(on=PleaseRetryError)
@raise_retry_on(return_value=0)
def f():
    ...
```

**Reply by SysuJayce** [reply]
Yeah I think it is a good alternative by raising an exception manually. Thank you @hynek
